### ISSUE TITLE
Equivalent of `jax.lax.fori_loop`

### ISSUE THREAD

**Issue by AlexanderMath** [onnx/onnx#6571]
# Ask a Question
When exporting a PyTorch model with `torch.onnx.export`, it seems there is no function like `jax.lax.fori_loop` such that 
```
for block in blocks: 
    x = block(x)
````
would be translated into an ONNX [Loop](https://onnx.ai/onnx/operators/onnx__Loop.html) operation. Is there a way I can do this manually by working with the ONNX file?

Why I care about this:  The time to create a session with the resulting ONNX model increasing linearly with `len(blocks)` from <1s to 10s.  A hacky fix, would be to export `block(x, weights)` and move the forloop out of the ONNX model - but ideally I'd be able to do this inside the ONNX model.

**Comment by justinchuby** [comment]
Yes. You can consider using the onnx IR https://aka.ms/onnx-ir or graph surgeon https://docs.nvidia.com/deeplearning/tensorrt/onnx-graphsurgeon/docs/index.html for manipulating ONNX graphs.

**Comment by justinchuby** [comment]
You can also use a while loop https://github.com/pytorch/pytorch/blob/main/torch/_higher_order_ops/while_loop.py
